src/charm.py

```diff
@@ -92,6 +93,41 @@ def __init__(self, *args):
             self._configure,
         )
 
+    def _on_collect_unit_status(self, event: CollectStatusEvent):
+        """Check the unit status and set to Unit when CollectStatusEvent is fired.
+
+        Args:
+            event: CollectStatusEvent
+        """
+        if invalid_configs := self._get_invalid_configs():
+            event.add_status(BlockedStatus(f"Configurations are invalid: {invalid_configs}"))
+            logger.info(f"Configurations are invalid: {invalid_configs}")
+            return
+        if not self._relation_created(N2_RELATION_NAME):
+            event.add_status(BlockedStatus("Waiting for N2 relation to be created"))
+            logger.info("Waiting for N2 relation to be created")
+            return
+        if not self._container.can_connect():
+            event.add_status(WaitingStatus("Waiting for container to be ready"))
+            logger.info("Waiting for container to be ready")
+            return
+        if not self._container.exists(path=BASE_CONFIG_PATH):
+            event.add_status(WaitingStatus("Waiting for storage to be attached"))
+            logger.info("Waiting for storage to be attached")
+            return
+        if not self._kubernetes_multus.multus_is_available():
+            event.add_status(BlockedStatus("Multus is not installed or enabled"))
+            logger.info("Multus is not installed or enabled")
+        if not self._kubernetes_multus.is_ready():
+            event.add_status(WaitingStatus("Waiting for Multus to be ready"))
+            logger.info("Waiting for Multus to be ready")
+            return
+        if not self._n2_requirer.amf_hostname or not self._n2_requirer.amf_port:
+            event.add_status(WaitingStatus("Waiting for N2 information"))
+            logger.info("Waiting for N2 information")
+            return
+        event.add_status(ActiveStatus())
```

Comment:
We should check that the upf config files are written and that the upf services are running before placing it in active